p2p/ipld/sample.go

```diff
@@ -0,0 +1,99 @@
+package ipld
+
+import (
+	"github.com/ipfs/go-cid"
+	"github.com/lazyledger/nmt/namespace"
+
+	"github.com/lazyledger/lazyledger-core/libs/rand"
+	"github.com/lazyledger/lazyledger-core/p2p/ipld/plugin/nodes"
+	"github.com/lazyledger/lazyledger-core/types"
+)
+
+// Sample is a point in 2D space over square.
+type Sample struct {
+	Row, Col uint32
+}
+
+// SampleSquare randomly picks *num* unique points from arbitrary *width* square
+// and returns them as samples.
+func SampleSquare(squareWidth uint32, num int) []*Sample {
```

Comment:
Do we need to add a panic here if num > squareWidth? It should never happen... but if it does won't this loop infinitely?

Reply:
In case we have `squareWidth = 2` and `num = 3` we would still have a valid situation. 

`squareWidth = 2` means we have 4 valid points, as the square is 2d space, and we can legally sample 3 of them.

Reply:
ahh I see. Yeah that's right. What about if `num > squareWidth*squareWidth`?

Reply:
Pushed the check